onnxruntime/core/providers/cuda/reduction/reduction_ops.cc

```diff
@@ -596,14 +596,26 @@ static Status ReduceComputeCore(CUDAExecutionProvider& cuda_ep, const Tensor& in
 template <bool allow_multi_axes>
 template <typename T, cudnnReduceTensorIndices_t ReduceTensorIndices>
 Status ReduceKernel<allow_multi_axes>::ComputeImpl(OpKernelContext* ctx, cudnnReduceTensorOp_t cudnn_reduce_op) const {
-  const Tensor* X = ctx->Input<Tensor>(0);
 
+  const std::string& op_name = this->KernelDef().OpName();
+  std::vector<int64_t> axes_values = axes_;
+  if (op_name == "ReduceSumTraining") {
+    //override the attribute value with the input value for reduction_axes
+    const Tensor* axesTensor = ctx->Input<Tensor>(1);
+    ORT_ENFORCE(axesTensor->Shape().NumDimensions() == 1, "An axes tensor must be a vector tensor.");
+    auto nDims = static_cast<size_t>(axesTensor->Shape()[0]);
```

Comment:
also need to handle negative axes

Reply:
already part of PrepareForReduce()

---
In reply to: [455327508](https://github.com/microsoft/onnxruntime/pull/4522#discussion_r455327508) [](ancestors = 455327508)